src/seals/diagnostics/base_env.py

```diff
@@ -0,0 +1,100 @@
+"""Minimal base class for simple environments."""
+
+import gym
+from gym.spaces import Discrete
+from gym.utils import seeding
+import numpy as np
+
+from seals.util import sample_distribution
+
+
+class BaseEnv(gym.Env):
```

Comment:
This overlaps a lot with https://github.com/HumanCompatibleAI/imitation/blob/c8d39a6e524466221935eec4e904dc9254bc2b96/src/imitation/envs/resettable_env.py -- I think we should combine them in some way.

@shwang how would you feel about making `seals` a required dependency of `imitation`, but ensuring that we keep `seals` dependencies light (things like MuJoCo/ALE will be kept optional, like in Gym)? We could then have a `seals.common` that defines some base classes like this.

Reply:
Some abstractions that we could have as mixins:

- Resettable environments
- Transitions/rewards for tabular MDPs
- _Maybe_ some intermediate mixin that gives you an explicit successor distribution for each state? e.g. have a `get_successors(state, action)` function that enumerates all successors of the current state given the current action, along with their probability (or some symbolic representation of that distribution, like a Gaussian). This would be sufficient to implement, e.g., VI, while still being suitable to represent large factored MDPs and continuous MDPs.